### ISSUE TITLE
Headers duplicated if server under high-load

### ISSUE THREAD

**Issue by dalisoft** [uNetworking/uWebSockets.js#245]
Hi.
Thanks for library.

I got issue in my library issues, which describes this:
- When testing with `stream file`, under high-load headers duplicated (up to Infinity)
- When under high-load, even not `stream file`, headers duplicated (1-5x times)
- When testing with `wrk` this issue happens too
- If as client using `Safari`, issue happening less than `Chrome`

Related issue, [here](https://github.com/sifrr/sifrr/issues/280) and [here](https://github.com/dalisoft/nanoexpress/issues/63)

**Comment by alexhultman** [comment]
Can you provide a minimal example of the issue with uWS? I don't know what you do in your wrapper and I don't know about any headers.

**Comment by dalisoft** [comment]
Hi @alexhultman and sorry for delay.
Thanks for your fast response.
https://github.com/dalisoft/uwebsockets-headers-duplicate-demo
I made simple repo with steps how to reproduce this, i checked multiple times and bug really exists

**Comment by dalisoft** [comment]
Sample screenshot

![](http://joxi.ru/Drl1gxlsVpN15m?d=1)

**Comment by alexhultman** [comment]
It seems pipelining might be a bit broken when streaming,

I can trigger the issue with wrk and curl like so:

./wrk -s scripts/pipeline.lua http://localhost:9001

curl --verbose http://localhost:9001

It's not just the headers, its the entire head that is duplicated including HTTP line. Probably some bug with corking in this case, will look at it however no browser use pipelining so it's less of an issue right now.

**Comment by dalisoft** [comment]
Thanks. Yes, for me it's not bug currently, everything works nice, i just got issue in my lib, so i decided let you know about this, even not in priority

**Comment by alexhultman** [comment]
You don't need anything more than wrk to trigger the issue:

./wrk -c 1 -t 1 -s scripts/pipeline.lua http://localhost:9001

It just gives you nothing:

Running 10s test @ http://localhost:9001
  1 threads and 1 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     0.00us    0.00us   0.00us    -nan%
    Req/Sec     0.00      0.00     0.00      -nan%
  0 requests in 10.10s, 0.00B read
  Socket errors: connect 0, read 67289, write 0, timeout 0
Requests/sec:      0.00
Transfer/sec:       0.00B

That's one connection only. So the issue is basically - streaming with pipelining is probably broken, while everything else works.

**Comment by dalisoft** [comment]
Yes, everything else works fine, even streaming on basic usage works fine.

Thanks

**Comment by alexhultman** [comment]
This has never worked and is more of a feature than a bug. It doesn't affect any real-world usage and has no security implications. It's more of a corner case supporting a dead feature.

**Comment by dalisoft** [comment]
Then we can close bug?

**Comment by alexhultman** [comment]
No it's a good reminder for the future. Maybe it will get fixed at some point.

**Comment by dalisoft** [comment]
Ok, thanks
